pkg/store/employee/employee.go

```diff
@@ -50,11 +51,11 @@ func (s *store) Search(db *gorm.DB, filter SearchFilter, pagination model.Pagina
 	}
 
 	query = query.Preload("ProjectMembers", "deleted_at IS NULL").
-		Preload("ProjectMembers.Project").
+		Preload("ProjectMembers.Project", "deleted_at IS NULL").
 		Preload("ProjectMembers.Project.Heads").
-		Preload("EmployeePositions", "deleted_at IS NULL").
-		Preload("EmployeePositions.Position").
+		Preload("Positions", "deleted_at IS NULL").
 		Preload("Roles", "deleted_at IS NULL").
+		Preload("Stacks", "deleted_at IS NULL").
```

Comment:
I think it's not good, in case we delete in table employee_stacks, and the condition check deleted_at in stacks table

Reply:
```
Preload("Stacks", "deleted_at IS NULL").
```
This function only checks with the deleted_at field in the `stacks` table.
Because we have many2many relations between `employees` and `stacks`, we need to check the field in `employee_stacks` table with deleted_at. Stacks is the definition data, so it's rarely modified.

Reply:
The same with `Preload("Roles", "deleted_at IS NULL").`

Reply:
updated